hazelcast/src/hazelcast/util/concurrent/CancellationException.cpp

```diff
@@ -26,6 +26,10 @@ namespace hazelcast {
             void CancellationException::raise() const {
                 throw *this;
             }
+
+            int32_t CancellationException::getErrorCode() const {
+                return client::protocol::CANCELLATION;
```

Comment:
We are providing the error code in the constructor as client::protocol::ILLEGAL_STATE. Then we are overriding it with this method. Can't we just pass `CANCELLATION` in the constructor to base class and remove `virtual` from IException?

Reply:
OK, changed as suggested.